src/js/port_handler.js

```diff
@@ -15,8 +15,24 @@ const PortHandler = new function () {
 };
 
 PortHandler.initialize = function () {
+    function generateVirtualApiVersions() {
```

Comment:
API versions do not mean much to most users, so this is not really helpful. I think this will be a lot easier to use if we add a mapping to firmware versions. This can probably be done as `1.43` => '4.2.0 and newer', and leave API versions where the version is not known (i.e. too old / new).